modules/dev-tools/ts-transform-append-extension/index.cjs

```diff
@@ -0,0 +1,51 @@
+/**
+ * Adapted from https://github.com/murolem/ts-transformer-append-js-extension
+ */
+const path = require('path');
+
+module.exports = function (program, pluginConfig, { ts }) {
```

Comment:
Maybe name the function and add some JSDoc describing what this does and why we forked, to guide the maintainer?

Reply:
I assume ESM plugins are not yet supported...

Reply:
ESM support is experimental, and requires additional dependencies

Reply:
Moved to .ts and added a build step